svthalia/Reaxit
# Introduce GoRouter

Closes #147.

### Summary
Replace ThaliaRouterDelegate with the go_router package. This makes it easy to set custom transitions per page, and offers support for having routing with stateful widgets: routing with tabs, or routing to screens within a single widget for correctly animating the menu drawer.

### How to test
- Try out push notifications on both iOS and android. Specifically pushnotifications without url's should cause dialogs to pop up, and notifications should be displayed even while the app is in the foreground.
- Try out deeplinks / universal links to various screens. Also test what happens when not logged in.
- Navigate everywhere from everywhere, to make sure all navigation buttons are correct.
